paas_app_charmer/fastapi/charm.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""FastAPI Charm service."""
+
+import pathlib
+import typing
+
+import ops
+from pydantic import BaseModel, Extra, Field
+
+from paas_app_charmer.app import App, WorkloadConfig
+from paas_app_charmer.charm import PaasCharm
+
+
+class FastAPIConfig(BaseModel, extra=Extra.allow):
+    """Represent FastAPI builtin configuration values.
+
+    Attrs:
+        uvicorn_port: port where the application is listening
```

Comment:
Should we use the `webserver` prefix here instead of `uvicorn`? Or did we use `hunicorn` for the comparable others?

Reply:
the attribute is `uvicorn_port: int = Field(alias="webserver-port", default=8080, gt=0)`, so in `charmcraft.yaml` the configuration option name is `webserver-port`. However, the environment variable has to be "UVICORN_PORT" (see https://www.uvicorn.org/settings/).

Same for uvicorn_port, uvicorn_host and web_concurrency.